src/utils/utils.py

```diff
@@ -194,17 +192,53 @@ def get_metric_value(metric_dict: dict, metric_name: str) -> float:
     return metric_value
 
 
-def close_loggers() -> None:
+def close_loggers(loggers: List[Logger]) -> None:
     """Makes sure all loggers closed properly (prevents logging failure during multirun)."""
 
     log.info("Closing loggers...")
 
-    if find_spec("wandb"):  # if wandb is installed
-        import wandb
-
-        if wandb.run:
+    # Lazily importing for efficiency
+    from pytorch_lightning.loggers.comet import CometLogger
+    from pytorch_lightning.loggers.csv_logs import CSVLogger
+    from pytorch_lightning.loggers.logger import DummyLogger
+    from pytorch_lightning.loggers.mlflow import MLFlowLogger
+    from pytorch_lightning.loggers.neptune import NeptuneLogger
+    from pytorch_lightning.loggers.tensorboard import TensorBoardLogger
+    from pytorch_lightning.loggers.wandb import WandbLogger
+
+    _AIM_AVAILABLE: bool = bool(RequirementCache("aim"))
+    if _AIM_AVAILABLE:
+        from aim.sdk.adapters.pytorch_lightning import AimLogger
+    else:
+        AimLogger = DummyLogger
+
+    closed: int = 0
+    for logger in loggers:
+        if isinstance(logger, DummyLogger):
+            pass
+        elif isinstance(logger, WandbLogger):
             log.info("Closing wandb!")
+            import wandb  # wandb guaranteed to exist here
+
             wandb.finish()
+            closed += 1
+        elif isinstance(logger, CometLogger):
+            pass
+        elif isinstance(logger, MLFlowLogger):
+            pass
+        elif isinstance(logger, NeptuneLogger):
+            pass
+        elif isinstance(logger, TensorBoardLogger):
+            pass
+        elif isinstance(logger, CSVLogger):
+            pass
+        elif isinstance(logger, AimLogger):
+            log.info(f"Closing aim run {logger._run_hash}")
+            logger._run.close()
+            del logger._run
+            logger._run = None
+            closed += 1
+    log.info(f"{closed} logger(s) closed.")
```

Comment:
Is this change still necessary? I now noticed passing `object_dict["trainer"].loggers` to `close_loggers()` is problematic. 

Sometimes during multirun one of the runs might fail. In such case we want to close loggers to make sure the next run of multirun won't run into any logging issues. But it's not possible to pass `object_dict` to `close_loggers()` if the run failed, because the `object_dict` is returned from finished task, so it won't be available at that point.

Reply:
@tesfaldet sorry for the ping, just wanted to make sure you haven't missed this comment

Reply:
Yup, you're absolutely correct and I actually just bumped into that issue right now. I can implement a fix tomorrow!

Reply:
Okay, so I think it might just be completely unnecessary to have `close_loggers()` entirely, considering that Lightning already closes loggers after their respective runs have finished. That means by the time `close_logger` executes, all the loggers will most likely (I'm not 100% sure) be closed. I think this deserves some looking into that goes beyond the scope of this PR so I'll remove these changes :)

Reply:
Alright, just made the changes! I squashed the commits to clean things up.

Reply:
> I think it might just be completely unnecessary to have close_loggers() entirely, considering that Lightning already closes loggers after their respective runs have finished (via their finalize() callback). That means by the time close_logger executes, all the loggers will most likely (I'm not 100% sure) be closed.

Not sure if they haven't changed anything recently, but last time I checked closing logger was definitely needed for wandb. I believe this was due to the fact that lightning doesn't have a way of knowing when the logging is finished. For example, you might execute:
```python
trainer.fit(model, datamodule)
trainer.test(model, testdata1)
trainer.test(model, testdata2)
trainer.test(model, testdata3)
# ... and so on
```
All those tests are supposed to log to the same session/experiment (at least in case of wandb).

At which point does the lightning call the logger `finalize()` method? 

Even if `finalize()` was somehow executed when the whole python program finishes, I don't think this would work in the hydra multirun setup due to that fact that multirun is a single process (I think?)

Reply:
No you are absolutely correct for wandb, actually. I completely forgot about that unique case. They heavily imply that's what's needed to be done over here https://github.com/Lightning-AI/lightning/blob/66966b405ee158dc31cfe78afc789ec25fedb022/src/lightning/pytorch/loggers/wandb.py#L393-L399.

Nonetheless, I still think this is all irrelevant to this PR, so I've changed things back to how they were previously!